src/bpm/runc/adapter/adapter_test.go

```diff
@@ -498,6 +497,38 @@ var _ = Describe("RuncAdapter", func() {
 			})
 		})
 
+		Context("when the user requests a privileged container", func() {
+			BeforeEach(func() {
+				procCfg.Unsafe = &config.Unsafe{Privileged: true}
+			})
+
+			It("uses the root user", func() {
+				spec, err := runcAdapter.BuildSpec(logger, bpmCfg, procCfg, user)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(spec.Process.User).To(Equal(specs.User{}))
+			})
+
+			It("does not set seccomp", func() {
+				spec, err := runcAdapter.BuildSpec(logger, bpmCfg, procCfg, user)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(spec.Linux.Seccomp).To(BeNil())
+			})
+
+			It("does not restrict capabilities", func() {
+				spec, err := runcAdapter.BuildSpec(logger, bpmCfg, procCfg, user)
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedCapabilities := append(defaultPrivilegedCapabilities, "CAP_TAIN", "CAP_SAICIN")
```

Comment:
```
expectedCapabilities := append(defaultPrivilegedCapabilities ...
```

above line may modify defaultPrivilegedCapabilities in place so future ref will contain changed value (behaviour is not specified by append to be certain and may change from go to go).  `package adapter_test` is a very good idea. super sad to see it go -- tests shouldnt use same constants that code uses, as then it doesnt really test it.

Reply:
I think I'd rather change it from a variable to a function that returns a slice.

As to the `_test`, I've been exploring following [go standards](https://golang.org/pkg/testing/)  with regards to package naming.

Reply:
I think the test and code thing in the same package is worth exploring, and makes more sense in Go than something equivalent would in some other languages.

Hopwever, I definitely agree with @cppforlife about reusing the constants in the test... easy to accidentally end up with incorrectly passing tests later on with this pattern.

Reply:
Could you also add a comment as to why `CAP_TAIN` and `CAP_SAICIN` are expected to be added? As someone just reading this test I don't know what they are or why I should expect them when they're not in the defaults.